src/Util.php

```diff
@@ -242,7 +242,8 @@ public static function cast(int $asType, $value)
         // -> to a boolean indicating whether the number is non-zero
         if ($type & JsonBrowser::TYPE_NUMBER) {
             if ($asType & JsonBrowser::TYPE_INTEGER) {
-                return $value > \PHP_INT_MAX ? floor($value) : (int) floor($value);
+                $int = $value > 0 ? floor($value) : ceil($value);
+                return abs($int) > \PHP_INT_MAX ? $int : (int) $int;
```

Comment:
So it's possible to get a float back from your int cast? Would that be surprising to the calling code?